cg/meta/workflow/analysis.py

```diff
@@ -423,3 +428,31 @@ def parse_analysis(
         """Parses output analysis files"""
 
         raise NotImplementedError
+
+    def clean_analyses(self, case_id: str) -> None:
+        """Add a cleaned at date for all analyses related to a case."""
+        analyses: list = self.status_db.family(case_id).analyses
+        LOG.info(f"Adding a cleaned at date for case {case_id}")
+        for analysis_obj in analyses:
+            analysis_obj.cleaned_at = analysis_obj.cleaned_at or dt.datetime.now()
+            self.status_db.commit()
+
+    def clean_run_dir(self, case_id: str, yes: bool, case_path: Union[List[Path], Path]) -> int:
+        """Remove workflow run directory."""
+
+        try:
+            self.verify_case_path_exists(case_id=case_id)
+        except FileNotFoundError:
+            self.clean_analyses(case_id)
+
+        if yes or click.confirm(f"Are you sure you want to remove all files in {case_path}?"):
+            if case_path.is_symlink():
+                LOG.warning(
+                    f"Will not automatically delete symlink: {case_path}, delete it manually",
+                )
+                return EXIT_FAIL
+
```

Comment:
```suggestion
if case_path.is_dir():
```
This might be a little bit too much since the path should only be a directory and not a file, but maybe it can be specified here since `verify_case_path_exists` would work even if the path was a file path instead of a directory

Reply:
I see what you mean, but either way, we want to delete it, right? Or is there a scenario you know where get_case_path will return a file instead of a directory?

Reply:
I don't know of any scenario where the path is a file path (I think this should not happen). But in the case of it being a file `rmtree` will not remove it. It would raise an error, but since `ignore_errors=True`, it will just exit without deleting the file.

Reply:
Great point, this should maybe be addressed in a future PR to handle errors related to rmtree better!